code/game/objects/structures/mirror.dm

```diff
@@ -9,74 +9,110 @@
 	var/shattered = 0
 
 
-/obj/structure/mirror/attack_hand(mob/user as mob)
+/obj/structure/mirror/attack_hand(mob/user)
+	choose(user, user)
+
+/obj/structure/mirror/proc/can_use(mob/living/user, mob/living/carbon/human/target)
 	if(shattered)
+		return FALSE
+	if(!ishigherbeing(user) || !ishuman(target))
+		return FALSE
+	if(!isturf(user.loc) || !isturf(target.loc))
+		return FALSE
+	if(!Adjacent(user) || !Adjacent(target))
+		return FALSE
+	if(user.incapacitated())
+		return FALSE
+	return TRUE
+
+/obj/structure/mirror/proc/choose(mob/living/user, mob/living/carbon/human/target)
+	if(!can_use(user, target))
 		return
-
-	if(ishuman(user))
-		var/mob/living/carbon/human/H = user
-		var/datum/role/vampire/V = isvampire(H)
-		if(isvampire(H))
-			if(!(VAMP_MATURE in V.powers))
+	var/mob/living/carbon/human/H = user
+	var/datum/role/vampire/V = isvampire(H)
+	if(V && !(VAMP_MATURE in V.powers))
+		to_chat(H, "<span class='notice'>You don't see anything.</span>")
+		return
+	if(user.hallucinating())
+		switch(rand(1,100))
+			if(1 to 20)
+				to_chat(H, "<span class='sinister'>You look like [pick("a monster","a goliath","a catbeast","a ghost","a chicken","the mailman","a demon")]! Your heart skips a beat.</span>")
+				H.Knockdown(4)
+				H.Stun(4)
+				return
+			if(21 to 40)
+				to_chat(H, "<span class='sinister'>There's [pick("somebody","a monster","a little girl","a zombie","a ghost","a catbeast","a demon")] standing behind you!</span>")
+				H.audible_scream()
+				H.dir = turn(H.dir, 180)
+				return
+			if(41 to 50)
 				to_chat(H, "<span class='notice'>You don't see anything.</span>")
 				return
-		if(user.hallucinating())
-			switch(rand(1,100))
-				if(1 to 20)
-					to_chat(H, "<span class='sinister'>You look like [pick("a monster","a goliath","a catbeast","a ghost","a chicken","the mailman","a demon")]! Your heart skips a beat.</span>")
-					H.Knockdown(4)
-					H.Stun(4)
+
+	var/which = alert(user, "What would you like to change?", "Appearance", "Hair", "Beard", "Undies")
+
+	if(!which || !can_use(user, target))
```

Comment:
this is the second appearance of can_use without any delay between, it seems impossible that this could be false (see L29)